src/torchmetrics/image/qnr.py

```diff
@@ -0,0 +1,226 @@
+    """Compute Quality with No Reference (QualityWithNoReference_) also now as QNR.
+
+    The metric is used to compare the joint spectral and spatial distortion between two images.
+
+    As input to ``forward`` and ``update`` the metric accepts the following input
+
+    - ``preds`` (:class:`~torch.Tensor`): High resolution multispectral image of shape ``(N,C,H,W)``.
+    - ``target`` (:class:`~Dict`): A dictionary containing the following keys:
+        - ``ms`` (:class:`~torch.Tensor`): Low resolution multispectral image of shape ``(N,C,H',W')``.
+        - ``pan`` (:class:`~torch.Tensor`): High resolution panchromatic image of shape ``(N,C,H,W)``.
+        - ``pan_lr`` (:class:`~torch.Tensor`): Low resolution panchromatic image of shape ``(N,C,H',W')``.
+
+    where H and W must be multiple of H' and W'.
+
+    As output of `forward` and `compute` the metric returns the following output
+
+    - ``qnr`` (:class:`~torch.Tensor`): if ``reduction!='none'`` returns float scalar tensor with average QNR value
+      over sample else returns tensor of shape ``(N,)`` with QNR values per sample
+
+    Args:
+        alpha: Relevance of spectral distortion.
+        beta: Relevance of spatial distortion.
+        norm_order: Order of the norm applied on the difference.
+        window_size: Window size of the filter applied to degrade the high resolution panchromatic image.
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        kwargs: Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+
+    Example:
+        >>> import torch
+        >>> _ = torch.manual_seed(42)
+        >>> from torchmetrics.image import QualityWithNoReference
+        >>> preds = torch.rand([16, 3, 32, 32])
+        >>> target = {
+        ...     'ms': torch.rand([16, 3, 16, 16]),
+        ...     'pan': torch.rand([16, 3, 32, 32]),
+        ... }
+        >>> qnr = QualityWithNoReference()
+        >>> qnr(preds, target)
+        tensor(0.9694)
+
+    """
+
+    higher_is_better: bool = True
+    is_differentiable: bool = True
+    full_state_update: bool = False
+    plot_lower_bound: float = 0.0
+    plot_upper_bound: float = 1.0
+
+    preds: List[Tensor]
+    ms: List[Tensor]
+    pan: List[Tensor]
+    pan_lr: List[Tensor]
+
+    def __init__(
+        self,
+        alpha: float = 1,
+        beta: float = 1,
+        norm_order: int = 1,
+        window_size: int = 7,
+        reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        rank_zero_warn(
+            "Metric `QualityWithNoReference` will save all targets and"
+            " predictions in buffer. For large datasets this may lead"
+            " to large memory footprint."
```

Comment:
```suggestion
            "Metric `QualityWithNoReference` will save all targets and predictions in buffer."
            " For large datasets this may lead to large memory footprint."
```